src/main/java/htsjdk/tribble/AbstractFeatureReader.java

```diff
@@ -169,25 +191,30 @@ public static boolean hasBlockCompressedExtension (final URI uri) {
      *
      * @return the header object we've read-in
      */
+    @Override
     public Object getHeader() {
         return header.getHeaderValue();
     }
 
     static class EmptyIterator<T extends Feature> implements CloseableTribbleIterator<T> {
-        public Iterator iterator() { return this; }
-        public boolean hasNext() { return false; }
-        public T next() { return null; }
-        public void remove() { }
+        @Override public Iterator<T> iterator() { return this; }
+        @Override public boolean hasNext() { return false; }
+        @Override public T next() { return null; }
+        @Override public void remove() { }
         @Override public void close() { }
     }
 
+    public static boolean isTabix(String resourcePath, String indexPath) throws IOException {
+        if(indexPath == null){
+            indexPath = ParsingUtils.appendToPath(resourcePath, TabixUtils.STANDARD_INDEX_EXTENSION);
+        }
+        return hasBlockCompressedExtension(resourcePath) && ParsingUtils.resourceExists(indexPath);
+    }
+
     public static class ComponentMethods{
```

Comment:
What is the point of this `ComponentMethods` inner class if you're just going to move the implementation to the enclosing class?